cpp/include/cudf/interop.hpp

```diff
@@ -395,22 +395,20 @@ unique_device_array_t to_arrow_host(
  * @param mr    Device memory resource used to allocate `cudf::table`
  * @return cudf table generated from given arrow Table
  */
-[[deprecated]] std::unique_ptr<table> from_arrow(
+[[deprecated("Use cudf::from_arrow_host")]] std::unique_ptr<table> from_arrow(
   arrow::Table const& input,
   rmm::cuda_stream_view stream      = cudf::get_default_stream(),
   rmm::device_async_resource_ref mr = rmm::mr::get_current_device_resource());
 
 /**
  * @brief Create `cudf::scalar` from given arrow Scalar input
  *
- * @deprecated Since 24.08.
- *
  * @param input `arrow::Scalar` that needs to be converted to `cudf::scalar`
  * @param stream CUDA stream used for device memory operations and kernel launches
  * @param mr    Device memory resource used to allocate `cudf::scalar`
  * @return cudf scalar generated from given arrow Scalar
  */
-[[deprecated]] std::unique_ptr<cudf::scalar> from_arrow(
+std::unique_ptr<cudf::scalar> from_arrow(
```

Comment:
This function doesn't have a matching replacement in `from_arrow_host`, so we can't deprecate it because there's nothing the user can do to change their code to avoid the deprecation warning.

Reply:
Reference: https://github.com/rapidsai/cudf/pull/16297#issuecomment-2238042514
We may have over deprecated here. Not a big deal. We can fix the scalar one in 24.10.

Reply:
Apologies for the trouble. I posted https://github.com/rapidsai/cudf/pull/16297#discussion_r1689993279 for context. I didn't notice the case where a now deprecated API was being called internally. @zeroshade is going to work on replacing the Python calls to `to_arrow` from the Python side, which was the only reason the scalar overload was added in the first place, so this is functionality that can be removed altogether. Please take a look at my comment and the linked PR from there and let me know if you see any issues (aside from the immediate disruption that you've now worked around, sorry again).

Reply:
OK, having read the background, this one can stay deprecated, but the docstring and message should be updated to point users to the alternative. @vyasr, I think you understand what that is better than me, so could you please suggest a wording?

Reply:
I'd rather remove the deprecation than ship the release with compile warnings.
We can deprecate this in 24.10 where calls to it are to be removed.

Reply:
The proposed replacement would be to call arrow's [`makeArrayFromScalar` factory](https://arrow.apache.org/docs/cpp/api/array.html#_CPPv419MakeArrayFromScalarRK6Scalar7int64_tP10MemoryPool) and then call [`exportArray`](https://arrow.apache.org/docs/cpp/api/c_abi.html#_CPPv411ExportArrayRK5ArrayP10ArrowArrayP11ArrowSchema) on that object. The second bit is also required for people trying to switch from the old `from_arrow(arrow::Array)` API as well so we could add that there if we wanted.

Reply:
We could then do `from_arrow_host` and get a `cudf::scalar`? Or is this really a function that we only introduced for convenience in pylibcudf and it should really go away completely.

Reply:
We really only introduced it from Python. So we would have to do a `get_element` on the resulting column to get a cudf scalar out of it, which IMO is fine. There is no way to do a 0-copy conversion of a scalar because there is no ABI-stable equivalent that can be passed around right now anyway.